daemon/cluster/cluster.go

```diff
@@ -334,8 +334,9 @@ func (c *Cluster) Cleanup() {
 		c.mu.Unlock()
 		return
 	}
-	defer c.mu.Unlock()
 	state := c.currentNodeState()
+	c.mu.Unlock()
```

Comment:
I've removed the latest commit in response to this comment. I think holding the lock here for an extra function call is slightly less confusing than acquiring the lock and releasing it right away. However if people think the alternative would be clearer or safer, I'm happy to switch to that.